File: 1658-Minimum Operations to Reduce X to Zero.py
class Solution(object):
    def minOperations(self, nums, x):
        """
        :type nums: List[int]
        :type x: int
        :rtype: int
        """
# sliding window
        summ = sum(nums)
        if summ < x:
            return -1
        
        curr = 0
        res = -1
        left = 0
        for i in range(len(nums)):
            curr += nums[i]
            while curr > summ-x:
                curr -= nums[left]
                left += 1
            if curr == summ-x:
                res = max(res, i-left+1 )
                
        return len(nums) - res if res!=-1 else -1

File: C++/1658-Minimum Operations to Reduce X to Zero.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int x) {
        int tot = accumulate(begin(nums), end(nums), 0), N = nums.size();
        int windowVal = tot-x;
        if (windowVal < 0) return -1;
        
        int res = -1, left = 0, curr = 0;

        for (int i=0; i<nums.size(); i++) {
            curr += nums[i];
            while (curr > windowVal) {
                curr -= nums[left];
                left++;
            }

            if (curr == windowVal) {
                res = max(res, i-left+1);
            }
        }

        return res == -1?-1:N-res;
    }
};